core/include/drivers/stm32_rif.h

```diff
@@ -79,26 +62,49 @@ struct rif_conf_data {
 	uint32_t *lock_conf;
 };
 
-/* This function checks if execution context is TDCID or not.
+#ifdef CFG_STM32_RIF
+static inline bool stm32_rif_scid_ok(uint32_t cidcfgr, uint32_t scid_m,
+				     uint32_t cid)
+{
+	if ((cidcfgr & scid_m) == SHIFT_U32(cid, SCID_SHIFT) &&
+	    !(cidcfgr & _CIDCFGR_SEMEN))
+		return true;
+
+	return false;
+}
+
+static inline bool stm32_rif_semaphore_enabled_and_ok(uint32_t cidcfgr,
+						      uint32_t wl)
+{
+	if ((cidcfgr & _CIDCFGR_CFEN) && (cidcfgr & _CIDCFGR_SEMEN) &&
+	    (cidcfgr & _CIDCFGR_SEMWL(wl)))
+		return true;
+
+	return false;
+}
+
+static inline bool stm32_rif_semaphore_mode_incorrect(uint32_t cidcfgr)
+{
+	if (!(cidcfgr & _CIDCFGR_CFEN) || !(cidcfgr & _CIDCFGR_SEMEN) ||
+	    (!(cidcfgr & _CIDCFGR_SEMWL(RIF_CID1)) &&
+	     (cidcfgr & _CIDCFGR_SEMEN)))
+		return true;
+
+	return false;
+}
+
+/* stm32_rifsc_check_tdcid - This function checks if the execution context is
+ *			     TDCID or not.
+ *
  * @tdcid_state: [out] Set to true if TDCID, false otherwise.
  * Returns TEE_ERROR_DEFER_DRIVER_INIT if RIFSC driver isn't probed, TEE_SUCCESS
  * otherwise.
  */
-#ifdef CFG_STM32_RIF
 TEE_Result stm32_rifsc_check_tdcid(bool *tdcid_state);
-#else
-static inline TEE_Result stm32_rifsc_check_tdcid(bool *tdcid_state)
-{
-	/* Without CFG_STM32_RIF every CPU can behave as TDCID */
-	*tdcid_state = true;
-
-	return TEE_SUCCESS;
-}
-#endif
 
 /*
- * Check every possible configuration where accessing RIF
- * is possible :
+ * stm32_rif_check_access - Check every possible configuration where accessing
```

Comment:
Append parentheses to function label
 s/stm32_rif_check_access/stm32_rif_check_access()/

Ditto at line 129.